client/js/keybinds.js

```diff
@@ -78,21 +79,8 @@ for (const hotkey in colorsHotkeys) {
 	Mousetrap.bind("mod+" + hotkey, function(e) {
 		e.preventDefault();
 
-		const cursorPosStart = input.prop("selectionStart");
-		const cursorPosEnd = input.prop("selectionEnd");
-		const value = input.val();
-		let newValue = value.substring(0, cursorPosStart) + colorsHotkeys[e.key];
-
-		if (cursorPosStart === cursorPosEnd) {
-			// If no text is selected, insert at cursor
-			newValue += value.substring(cursorPosEnd, value.length);
-		} else {
-			// If text is selected, insert formatting character at start and the end
-			newValue += value.substring(cursorPosStart, cursorPosEnd) + colorsHotkeys[e.key] + value.substring(cursorPosEnd, value.length);
-		}
-
-		input
-			.val(newValue)
-			.get(0).setSelectionRange(cursorPosStart + 1, cursorPosEnd + 1);
+		const modifier = colorsHotkeys[e.key];
+
+		wrapCursor(input, modifier, input.selectionStart === input.selectionEnd ? "" : modifier);
```

Comment:
Ok so just making sure I read this correctly:

- If there is text selected, add `modifier` characters around that selection.
- If there is nothing selected, effectively "wrap" `modifier` on the left and empty string on the right.

Also if I'm reading this correctly: intuitively it seems like the second case should just be: if no text is selected, simply add character, but it's beneficial to re-use `wrapCursor` with that empty string trick because 1) it avoids tampering the string ourselves to insert the char, 2) whatever undo mechanism the first scenario is offered with the first scenario also applies in the second scenario.

Does that sound correct to you?

Reply:
> If there is nothing selected, effectively "wrap" modifier on the left and empty string on the right.

That's essentially right. I had to add this selection check because that lib *always* wraps around cursor (selection) position.

Reply:
:+1: :shipit: